### PR TITLE
Inactive movements fix

### PR DESCRIPTION
**Pull request by rperez89**
Fix for [notion item 3](https://www.notion.so/Inactive-24h-message-is-the-same-as-the-active-wallet-message-it-shouldn-t-show-anything-if-the-ba-3f194d7022964b5d979f504d5f9f0359)

### CODE REVIEW

File: src/hooks/subscription-hooks.js [hunk 1]
```diff
@@ -76,6 +77,33 @@ export function useJurorBalancesSubscription(jurorId) {
...
+    const movementsFiltered = movements => {
+      // If the juror is activating from the wallent we need to just show an incoming movement in the Active balance
+      return movements.reduce((previousList, movement, index, originalList) => {
+        const convertedMovement = {
+          ...movement,
+          effectiveTermId: parseInt(movement.effectiveTermId, 10),
+          amount: bigNum(movement.amount),
+        }
+        if (index === 0) {
+          previousList[0] = convertedMovement
```
Comment (fabriziovigevani) [inline src/hooks/subscription-hooks.js]: Here we could just do push()

```suggestion
          previousList.push(convertedMovement)
```

File: src/hooks/subscription-hooks.js [hunk 2]
```diff
@@ -76,6 +77,33 @@ export function useJurorBalancesSubscription(jurorId) {
       movements = [],
     } = jurorData.juror || {}
 
+    const movementsFiltered = movements => {
+      // If the juror is activating from the wallent we need to just show an incoming movement in the Active balance
+      return movements.reduce((previousList, movement, index, originalList) => {
```
Comment (fabriziovigevani) [inline src/hooks/subscription-hooks.js]: Should we rename `previousList` to something with more meaning? 
Was thinking something like `accumulatorList`

File: src/hooks/subscription-hooks.js [hunk 3]
```diff
@@ -76,6 +77,33 @@ export function useJurorBalancesSubscription(jurorId) {
...
+          effectiveTermId: parseInt(movement.effectiveTermId, 10),
+          amount: bigNum(movement.amount),
+        }
+        if (index === 0) {
+          previousList[0] = convertedMovement
+        } else if (
+          ANJMovement[originalList[index - 1].type] ===
+            ANJMovement.Activation &&
+          ANJMovement[movement.type] === ANJMovement.Stake &&
+          originalList[index - 1].createdAt === movement.createdAt
```
Comment (fabriziovigevani) [inline src/hooks/subscription-hooks.js]: What do you think of storing a variable for `originalList[index - 1]` like `previousMovement` and use it here so it's more clear?

File: src/hooks/subscription-hooks.js [hunk 4]
```diff
@@ -76,6 +77,33 @@ export function useJurorBalancesSubscription(jurorId) {
...
+        ) {
+          previousList[previousList.length - 1] = {
+            ...previousList[previousList.length - 1],
+            type: 'StakeActivation',
+          }
+        } else {
+          previousList[previousList.length] = convertedMovement
```
Comment (fabriziovigevani) [inline src/hooks/subscription-hooks.js]: Same here we can do `.push()`
```suggestion
          previousList.push(convertedMovement)
```

File: src/hooks/subscription-hooks.js [hunk 5]
```diff
@@ -76,6 +77,33 @@ export function useJurorBalancesSubscription(jurorId) {
...
+          previousList[0] = convertedMovement
+        } else if (
+          ANJMovement[originalList[index - 1].type] ===
+            ANJMovement.Activation &&
+          ANJMovement[movement.type] === ANJMovement.Stake &&
+          originalList[index - 1].createdAt === movement.createdAt
+        ) {
+          previousList[previousList.length - 1] = {
+            ...previousList[previousList.length - 1],
+            type: 'StakeActivation',
```
Comment (fabriziovigevani) [inline src/hooks/subscription-hooks.js]: If we move this function to the file we could also store `StakeActivation` in a const variable

File: src/hooks/subscription-hooks.js (hunk 2 above)
Comment (bpierre) [reply src/hooks/subscription-hooks.js]: I think we could use only one condition (also taking your suggestions @fabriziovigevani):

```jsx
  return movements.reduce((movements, movement, index, prevMovements) => {
    if (
      index > 0 &&
      ANJMovement[prevMovements[index - 1].type] === ANJMovement.Activation &&
      ANJMovement[movement.type] === ANJMovement.Stake &&
      prevMovements[index - 1].createdAt === movement.createdAt
    ) {
      movements[movements.length - 1].type = 'StakeActivation'
      return movements
    }

    movements.push({
      ...movement,
      effectiveTermId: parseInt(movement.effectiveTermId, 10),
      amount: bigNum(movement.amount),
    })

    return movements
  })

```

**Pull request merged**